lib/SPIRV/SPIRVReader.cpp

```diff
@@ -1552,14 +1552,14 @@ Value *SPIRVToLLVM::transValueWithoutDecoration(SPIRVValue *BV, Function *F,
         const unsigned RawDataNumWords = BConst->getNumWords();
         const unsigned BigValNumWords = (RawDataNumWords + 1) / 2;
         std::vector<uint64_t> BigValVec(BigValNumWords);
-        const SPIRVWord *RawData = BConst->getSPIRVWords();
+        const std::vector<SPIRVWord> &RawData = BConst->getBigValWords();
         // SPIRV words are integers of 32-bit width, meanwhile llvm::APInt
         // is storing data using an array of 64-bit words. Here we pack SPIRV
         // words into 64-bit integer array.
-        for (size_t I = 0; I != RawDataNumWords; ++I)
-          BigValVec[I / 2] =
-              (I % 2) ? BigValVec[I / 2] | ((uint64_t)RawData[I] << 32)
-                      : BigValVec[I / 2] | ((uint64_t)RawData[I]);
+        for (size_t I = 0; I != RawDataNumWords / 2; ++I)
+          BigValVec[I] = ((uint64_t)RawData[2 * I + 1] << 32) | RawData[2 * I];
+        if (RawDataNumWords % 2)
+          BigValVec[BigValNumWords - 1] = RawData[RawDataNumWords - 1];
```

Comment:
```suggestion
          BigValVec.back() = RawData.back();
```

I'm not sure about the `RawData.back()` though (I just assume you are looking for the last item in the vector)

Reply:
Thank you for a good catch. Fixed.